From f38b5c9837476af0b486e88f9d86d88580feb05a Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Mon, 25 Mar 2024 12:15:21 +0100
Subject: GCC: make base::span::as_chars a constexpr

Quite often, base::span::as_chars is called from constexpr, and
warnings are emitted by GCC that it is not a constexpr. So add the
keyword accordingly.

Bug: 40565911
Change-Id: If620c3105f7119dcc6e893a7e70024685d1294a5
---
 base/containers/span.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/base/containers/span.h b/base/containers/span.h
index e62ce9163c740..e05bfa7f66698 100644
--- a/base/containers/span.h
+++ b/base/containers/span.h
@@ -1049,7 +1049,7 @@ span(const T (&)[N]) -> span<const T, N>;
 
 // [span.objectrep], views of object representation
 template <typename T, size_t X>
-auto as_bytes(span<T, X> s) noexcept {
+constexpr auto as_bytes(span<T, X> s) noexcept {
   constexpr size_t N = X == dynamic_extent ? dynamic_extent : sizeof(T) * X;
   // SAFETY: span provides that data() points to at least size_bytes() many
   // bytes. So since `uint8_t` has a size of 1 byte, the size_bytes() value is
@@ -1063,7 +1063,7 @@ auto as_bytes(span<T, X> s) noexcept {
 
 template <typename T, size_t X>
   requires(!std::is_const_v<T>)
-auto as_writable_bytes(span<T, X> s) noexcept {
+constexpr auto as_writable_bytes(span<T, X> s) noexcept {
   constexpr size_t N = X == dynamic_extent ? dynamic_extent : sizeof(T) * X;
   // SAFETY: span provides that data() points to at least size_bytes() many
   // bytes. So since `uint8_t` has a size of 1 byte, the size_bytes() value is a
@@ -1080,7 +1080,7 @@ auto as_writable_bytes(span<T, X> s) noexcept {
 // added since chrome still represents many things as char arrays which
 // rightfully should be uint8_t.
 template <typename T, size_t X>
-auto as_chars(span<T, X> s) noexcept {
+constexpr auto as_chars(span<T, X> s) noexcept {
   constexpr size_t N = X == dynamic_extent ? dynamic_extent : sizeof(T) * X;
   // SAFETY: span provides that data() points to at least size_bytes() many
   // bytes. So since `char` has a size of 1 byte, the size_bytes() value is a
-- 
2.40.1

